### PR TITLE
T35078 Get all nodes matching attributes from API

### PR DESCRIPTION
**Pull request by JenySadadia**
Implement KernelCI_API.get_nodes
to get all the nodes from API.

Used this in https://github.com/kernelci/kernelci-pipeline/pull/71

Signed-off-by: Jeny Sadadia <jeny.sadadia@collabora.com>

### CODE REVIEW

**Review by gctucker — COMMENTED** [review]
Sorry we can't do that, it will not scale and eventually break the server when we get to a real number of nodes.  We might be able to add pagination to receive nodes in chunks, but that's something we should be adding to the existing methods that already retrieve nodes with some filters.

**Comment by JenySadadia** [comment]
> Sorry we can't do that, it will not scale and eventually break the server when we get to a real number of nodes. We might be able to add pagination to receive nodes in chunks, but that's something we should be adding to the existing methods that already retrieve nodes with some filters.

Makes sense. I can change the method to get only pending nodes. That is what we need in the `set_timeout` script.

**Comment by gctucker** [comment]
We could also just have a `get_nodes()` method with some filters that would be passed to the `/nodes` endpoint.  We should also set a maximum number of nodes until we have support for paginated results.

**Comment by JenySadadia** [comment]
> We could also just have a `get_nodes()` method with some filters that would be passed to the `/nodes` endpoint. We should also set a maximum number of nodes until we have support for paginated results.

Updated the method. I will also send a PR to drop the `get_nodes_by_commit_hash()` method as now `get_nodes()` can be used to get nodes by commit hash as well.

**Comment by JenySadadia** [comment]
I think `get_child_nodes_from_parent` can be dropped too. What do you suggest? @gctucker

File: kernelci/db/kernelci_api.py [hunk 1]
```diff
@@ -88,6 +88,12 @@ def get_node(self, node_id):
         resp = self._get('/'.join(['node', node_id]))
         return json.loads(resp.text)
 
+    def get_nodes(self, attributes: dict = None):
+        """Get all nodes matching attributes"""
+        resp = self._get('nodes', params=attributes)
+        # Maximum 10 nodes will be returned
+        return resp.json()[:10]
```
Comment (gctucker) [inline kernelci/db/kernelci_api.py:95]: I think the limit should be an argument of the method, and ideally this should be implemented as pagination in the API. Then we should also have a `skip` argument to get the next "page" of nodes.

Comment (JenySadadia) [reply kernelci/db/kernelci_api.py]: Implemented pagination in the API https://github.com/kernelci/kernelci-api/pull/164.
`skip` and `limit` arguments can be sent from client scripts using `attributes` dictionary.

**Review by gctucker — APPROVED** [review]
Tested OK on staging.

Got the email report:
```
kernelci/staging-mainline staging-mainline-20220804.0: 2 runs 0 fails

Tests Summary
-------------

Tree:     kernelci
Branch:   staging-mainline
Describe: staging-mainline-20220804.0
URL:      https://github.com/kernelci/linux.git
SHA1:     0b109692b29381c4e3d9d1adff0d4f120ee78f84

test            | result
----------------+-------
kver            | pass
kunit           | pass
```

**Pull request merged**